File: WebApplication1/Contracts/IPeopleRepository.cs
﻿using ApiDtoLibrary.Notifications;
using ApiDtoLibrary.Person;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using WebApplication1.Services;
using WebApplication1.Services.People;

namespace WebApplication1.Contracts
{
    public interface IPeopleRepository
    {
        Task<Person> GetPerson(Guid id);
        List<Person> GetAllPersonalData();
        Task DeleteAsync(Guid id);
        Task<IList<Notification>> GetNotifications(string recipientId);
    }
}

File: FsUniversityApi/Contracts/IPeopleRepository.fs
namespace FsUniversityApi.Contracts

open FsUniversityApi.Database.Entities.Person
open System.Threading.Tasks
open System
open System.Collections.Generic

type IPeopleRepository =
    abstract member GetAsync : Guid -> Task<PersonInfo>
    abstract member GetAllPersonalDataAsync : unit -> Task<List<PersonInfo>>
    abstract member DeleteAsync : Guid -> Task